service-app/internal/api/handler.go

```diff
@@ -281,6 +327,40 @@ func (c *IndexController) validateAndSanitizeXML(bodyStr string) (*types.BaseSet
 		return nil, err
 	}
 
+	// Validate embedded documents
+	for _, document := range parsedBaseXml.Body.Documents {
```

Comment:
Entirely up to you, but I'd consider putting this in a helper for clarity.

Reply:
Split out and I added more testing around it too